doc/adr/0004-structure-convention.md

```diff
@@ -0,0 +1,68 @@
+# 4. Structure convention
+
+Date: 2024-09-24
+
+## Status
+
+Accepted
+
+## Context
+
+The core packages gains more and more logic, we need a way to have small piece of code that are easily maintainable.
+
+## Decision
+
+### Domain Driven
+Code is split by domain. Each domain will have its own directory, and within each directory, the code will be further organized by functionality. This structure will help in maintaining a clean and manageable codebase.
+
+### Naming Convention
+
+To maintain consistency across the codebase, we will adopt a clear and concise naming convention. All files and directories will use `kebab-case` for their names. Except for main or index file, each file will be named using the format `name-of-file.name-of-folder.ts`, ensuring uniformity and readability. 
```

Comment:
Naming, could you provide more examples on this? For instance, how would you name a service file or a utility file within a domain?